packages/uniforms/__tests__/AutoForm.tsx

```diff
@@ -99,85 +121,128 @@ describe('AutoForm', () => {
       }
 
       // FIXME: AutoForm is not a valid Component.
-      mount<CustomAutoForm | any>(
+      render(
         // @ts-expect-error Convoluted AutoForm types
         <CustomAutoForm
-          autoField={Field}
-          model={model}
           onChange={onChange}
           schema={schema}
+          autoField={Field}
+          model={model}
         />,
+        { schema: { type: SimpleSchema2Bridge } },
       );
 
       expect(onChange).toHaveBeenCalledTimes(2);
       expect(onChange.mock.calls[0]).toEqual(expect.arrayContaining(['b', '']));
       expect(onChange.mock.calls[1]).toEqual(expect.arrayContaining(['c', '']));
     });
-
     it('skips `onSubmit` until rendered (`autosave` = true)', async () => {
       // FIXME: AutoForm is not a valid Component.
-      const wrapper = mount<AutoForm | any>(
-        <AutoForm autosave onSubmit={onSubmit} schema={schema} />,
+      render(
+        <AutoForm autosave onSubmit={onSubmit} schema={schema}>
+          <AutoFields />
+        </AutoForm>,
+        {
+          schema: { type: SimpleSchema2Bridge },
+        },
       );
 
       expect(onSubmit).not.toBeCalled();
-      wrapper.instance().getContext().onChange('a', 1);
 
       await new Promise(resolve => setTimeout(resolve));
 
+      const input = screen.getByLabelText('A');
+
       expect(onSubmit).toHaveBeenCalledTimes(1);
-      expect(onSubmit).toHaveBeenLastCalledWith({ a: 1 });
-      expect(validator).toHaveBeenCalledTimes(1);
-      expect(validator).toHaveBeenLastCalledWith({ a: 1 });
+      expect(onSubmit).toHaveBeenLastCalledWith({ a: '', b: '', c: '' });
+
+      await new Promise(resolve => setTimeout(resolve));
+      fireEvent.change(input, { target: { value: '1' } });
+
+      expect(validator).toHaveBeenCalledTimes(2);
+      expect(validator).toHaveBeenLastCalledWith({ a: '1', b: '', c: '' });
     });
   });
 
   describe('when reset', () => {
     it('reset `model`', () => {
-      // FIXME: AutoForm is not a valid Component.
-      const wrapper = mount<AutoForm | any>(
-        <AutoForm autosave model={model} schema={schema} />,
+      const { rerenderWithProps } = render(
+        // FIXME: AutoForm is not a valid Component.
+        <AutoForm autosave model={model} schema={schema}>
+          <context.Consumer>
+            {context => mockContext(context?.model)}
+          </context.Consumer>
+        </AutoForm>,
+        {
+          schema: { type: SimpleSchema2Bridge },
+        },
       );
 
-      wrapper.instance().reset();
-      expect(wrapper.instance().getContext().model).toEqual(model);
+      rerenderWithProps({});
```

Comment:
Could we avoid calling `rerenderWithProps({})` if we don't want to override any props? Use `rerender`, this way we can skip the unnecessary `cloneElement` operation.